x/wasm/keeper/recursive_test.go

```diff
@@ -112,29 +104,29 @@ func TestGasCostOnQuery(t *testing.T) {
 		"no recursion, no work": {
 			gasLimit:    400_000,
 			msg:         Recurse{},
-			expectedGas: GasNoWork,
+			expectedGas: 0xa88d,
 		},
 		"no recursion, some work": {
 			gasLimit: 400_000,
 			msg: Recurse{
 				Work: 50, // 50 rounds of sha256 inside the contract
 			},
-			expectedGas: GasWork50,
+			expectedGas: 0xa9d8,
 		},
 		"recursion 1, no work": {
 			gasLimit: 400_000,
 			msg: Recurse{
 				Depth: 1,
 			},
-			expectedGas: 2*GasNoWork + GasReturnUnhashed,
+			expectedGas: 0x150f8,
```

Comment:
it's changed the test logic. Can you tell me where this change come from?

Reply:
The new gas calculation strategy in wasmvm leads to different values in the expected gas. This also happened in the wasmd update to v0.27. They needed to tinker with the expected test values too. I simply took the values that the test would expect.

I couldn't take the test values from wasmd. Because, the thing with terrad and wasmd is that the gas cost for emitted events from a contract call are inherently different (terrad charges per byte, while wasmd charges per event). That's why I kinda figured the expected test values are more or less regression reassurances for updates that come after this one.